### PR TITLE
[pegtl] Update to 3.2.1

### PR DESCRIPTION
**Pull request by chausner**
**Describe the pull request**

- #### What does your PR fix?  
  Updates pegtl to 3.2.1

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  As before, Yes

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes

- #### If you have added/updated a port: Have you run ./vcpkg x-add-version --all and committed the result?  
  Yes

### CODE REVIEW

File: ports/pegtl/vcpkg.json [hunk 1]
```diff
@@ -1,7 +1,16 @@
 {
   "name": "pegtl",
-  "version-string": "3.2.0",
-  "port-version": 1,
+  "version-semver": "3.2.1",
```
Comment (chausner) [inline ports/pegtl/vcpkg.json:3]: Follows semver according to https://github.com/taocpp/PEGTL#status.

**Review by JonLiu1993 — APPROVED** [review]

File: ports/pegtl/portfile.cmake [hunk 2]
```diff
@@ -1,26 +1,26 @@
...
-vcpkg_fixup_cmake_targets(CONFIG_PATH share/pegtl/cmake)
+vcpkg_cmake_config_fixup(CONFIG_PATH share/pegtl/cmake)
 
-file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug)
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug")
 
 # Handle copyright
-file(RENAME ${CURRENT_PACKAGES_DIR}/share/pegtl/LICENSE ${CURRENT_PACKAGES_DIR}/share/${PORT}/copyright)
+file(RENAME "${CURRENT_PACKAGES_DIR}/share/pegtl/LICENSE" "${CURRENT_PACKAGES_DIR}/share/${PORT}/copyright")
+
```
Comment (NancyLi1013) [inline ports/pegtl/portfile.cmake:26]:
```suggestion
```

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: versions/p-/pegtl.json [hunk 3]
```diff
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "9f1940d2f2fe494042528d38159260879648f16b",
```
Comment (NancyLi1013) [inline versions/p-/pegtl.json:4]:
```suggestion
      "git-tree": "4a95202decfce79a39190525dea1c3a135bf393f",
```

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Review by NancyLi1013 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for the update and semver research!

**Pull request merged**